### PR TITLE
Guns dual wielding

### PR DESCRIPTION
**Pull request by BombasterDS**
## About the PR
Most guns can be shoot from both hands now with lower accuracy

## Why / Balance
Funny SS13 TG mechanic (i like dual berettas from cs 1.6)

## Media

https://github.com/user-attachments/assets/97fef649-0ccd-475d-b7a5-7aaa3fd7a3ed

:cl:
- tweak: Dual welding for guns! Now you can shoot from most weapons with both hands!

## Summary by CodeRabbit

- **New Features**
	- Introduced a multishot capability for various weapons, allowing players to fire multiple projectiles simultaneously.
	- Added a `SpreadMultiplier` property to adjust projectile spread for enhanced tactical gameplay.
	- New entity types for multishot functionality added across multiple weapon categories including pistols, rifles, shotguns, and more.

- **Bug Fixes**
	- Improved ammunition management for multishot weapons to ensure accurate ammo counts after shooting.

- **Chores**
	- Updated weapon configuration files to include new multishot types and properties.

### CODE REVIEW

**Comment by whateverusername0** [comment]
the decreased accuracy from not wielding one gun already does the trick
you just won't be able to hit anything
also how does it work with pistols?

**Comment by BombasterDS** [comment]
> the decreased accuracy from not wielding one gun already does the trick you just won't be able to hit anything also how does it work with pistols?

Pistols have x1.2 multiplier to spread. Revolvers x1.1. (L6 x1.8 :godo)

File: Content.Shared/Goobstation/Weapons/Multishot/SharedMultishotSystem.cs [hunk 1]
```diff
@@ -64,7 +64,7 @@ private void OnEquipWeapon(Entity<MultishotComponent> multishotWeapon, ref GotEq
         if (!TryComp<HandsComponent>(args.User, out var handsComp))
             return;
 
-        if (handsComp.Count < 2 || handsComp.Count != comp.RequiredHandsCount)
+        if (handsComp.Count != 2)
```
Comment (whateverusername0) [inline Content.Shared/Goobstation/Weapons/Multishot/SharedMultishotSystem.cs:67]: and what if there's a species with 4 hands? :trollface:

**Comment by Piras314** [comment]
the speed of shooting should be slowed down significantly (like the guns should shoot at 50-60% speed so dual-wielding essentially makes it just less accurate, twice the mag size, and a bit faster shooting rather than just inaccurately shooting a ridiculous amount of bullets all at once

File: Content.Shared/Goobstation/Weapons/Multishot/SharedMultishotSystem.cs (hunk 1 above)
Comment (BombasterDS) [reply Content.Shared/Goobstation/Weapons/Multishot/SharedMultishotSystem.cs:67]: > and what if there's a species with 4 hands? :trollface:

i'll change it when they'll add species with 4 hands :trollface:

**Review by Piras314 — CHANGES_REQUESTED** [review]
why did you remove the spread multipliers from the shotguns

**Comment by BombasterDS** [comment]
> why did you remove the spread multipliers from the shotguns

Basic spread modifier is 1.5. Anyway, dual welding works only on sawed off and pirate pistol, because shotguns requires welding to use them.

**Pull request merged**